addons/website_event/models/website_visitor.py

```diff
@@ -82,16 +87,8 @@ def _inactive_visitors_domain(self):
         domain = super()._inactive_visitors_domain()
         return expression.AND([domain, [('event_registration_ids', '=', False)]])
 
-    def _link_to_partner(self, partner, update_values=None):
-        """ Propagate partner update to registration records """
-        if partner:
-            registration_wo_partner = self.event_registration_ids.filtered(lambda registration: not registration.partner_id)
-            if registration_wo_partner:
-                registration_wo_partner.partner_id = partner
-        super(WebsiteVisitor, self)._link_to_partner(partner, update_values=update_values)
-
-    def _link_to_visitor(self, target):
+    def _merge_visitor(self, target):
         """ Override linking process to link registrations to the final visitor. """
-        self.event_registration_ids.write({'visitor_id': target.id})
-
-        return super(WebsiteVisitor, self)._link_to_visitor(target)
+        self.event_registration_ids.visitor_id = target.id
+        self.event_registration_ids.partner_id = target.partner_id
```

Comment:
You should not update partner if there are already partner on registrations. We want to keep possibility of different partners on a given subset of registrations, even if linked to the same visitor. See old ``_link_to_partner`` code.